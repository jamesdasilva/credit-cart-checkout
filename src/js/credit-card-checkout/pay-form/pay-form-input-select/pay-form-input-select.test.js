import '@testing-library/jest-dom'
import { render, fireEvent } from '@testing-library/react'
import React from 'react'

import PayFormSelect from './pay-form-input-select'

describe('<PayFormSelect /> spec', () => {
  test('Snapshot test', () => {
    const _mock = {
      name: "installments",
      label: "Número de parcelas",
      invalidMsg: "Insira o número de parcelas",
      update: jest.fn()
    }
    const { getByTestId } = render(
      <PayFormSelect
        name={_mock.invalidMsg}
        label={_mock.label}
        invalidMsg={_mock.invalidMsg}
        update={_mock.update} />
      )

    const payFormSelect = getByTestId('pay-select')
    expect(payFormSelect).toMatchSnapshot()
  })
  test('Deve chamar update passando o novo valor quando houver alteração', () => {
    const _mock = {
      name: "installments",
      label: "Número de parcelas",
      invalidMsg: "Insira o número de parcelas",
      update: jest.fn()
    }
    const _paramCall = { 
      name: _mock.name, 
      value: '1', 
      isValid: true 
    }
    const { getByTestId } = render(
      <PayFormSelect
        name={_mock.name}
        label={_mock.label}
        invalidMsg={_mock.invalidMsg}
        update={_mock.update} />
      )

    const inputSelect = getByTestId('input-select')
    fireEvent.change(inputSelect, { target: { value: "1" } });
    expect(_mock.update.mock.calls[0][0]).toEqual(_paramCall)
  })

  test('Deve retornar { ...p, isValid: true } quando o valor for igual a 0', () => {
    const _mock = {
      name: "installments",
      label: "Número de parcelas",
      invalidMsg: "Insira o número de parcelas",
      update: jest.fn()
    }
    const _paramCall = { 
      name: _mock.name, 
      value: '0', 
      isValid: false 
    }
    const { getByTestId } = render(
      <PayFormSelect
        name={_mock.name}
        label={_mock.label}
        invalidMsg={_mock.invalidMsg}
        update={_mock.update} />
      )

    const inputSelect = getByTestId('input-select')
    fireEvent.change(inputSelect, { target: { value: "0" } });
    expect(_mock.update.mock.calls[0][0]).toEqual(_paramCall)
  })

  test('Deve retornar { ...p, isValid: true } quando o valor for diferente de 0', () => {
    const _mock = {
      name: "installments",
      label: "Número de parcelas",
      invalidMsg: "Insira o número de parcelas",
      update: jest.fn()
    }
    const _paramCall = { 
      name: _mock.name, 
      value: '3', 
      isValid: true 
    }
    const { getByTestId } = render(
      <PayFormSelect
        name={_mock.name}
        label={_mock.label}
        invalidMsg={_mock.invalidMsg}
        update={_mock.update} />
      )

    const inputSelect = getByTestId('input-select')
    fireEvent.change(inputSelect, { target: { value: "3" } });
    expect(_mock.update.mock.calls[0][0]).toEqual(_paramCall)
  })
})