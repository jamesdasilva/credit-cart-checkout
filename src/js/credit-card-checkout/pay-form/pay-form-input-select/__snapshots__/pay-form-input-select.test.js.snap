// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PayFormSelect /> spec Snapshot test 1`] = `
<div
  class="pay-select"
  data-testid="pay-select"
>
  <label
    class="pay-select__label"
  >
    Número de parcelas
  </label>
  <div
    class="pay-select__container pay-select__container--empty"
  >
    <select
      class="pay-select__select"
      data-testid="input-select"
      id="Insira o número de parcelas"
      placeholder="Número de parcelas"
    >
      <option
        value="0"
      >
        Número de parcelas
      </option>
      <option
        value="1"
      >
        1X
      </option>
      <option
        value="3"
      >
        3X
      </option>
      <option
        value="5"
      >
        5X
      </option>
      <option
        value="10"
      >
        10X
      </option>
    </select>
  </div>
  <div
    class="pay-select__alert pay-select__alert--invisible"
  >
    Insira o número de parcelas
  </div>
</div>
`;
